sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2230,6 +2230,8 @@ class Analyzer(
           val result = resolved transformDown {
             case UnresolvedMapObjects(func, inputData, cls) if inputData.resolved =>
               inputData.dataType match {
+                case ArrayType(et, false) if cls.isEmpty =>
```

Comment:
To be safe, we should check:

1. no custom collection class specified
2. the `function` will convert an expression `e` to `AssertNotNull(e)`(this guarantees we are expecting a primitive array)
3. the `inputData` is of type array and its element is not nullable.